File: module_my/src/main/java/com/jason/module_my/model/repository/DetailsRepository.java
package com.jason.module_my.model.repository;

import com.jason.module_my.config.Config;
import com.jason.module_my.http.ApiService;
import com.jason.module_my.model.bean.DetailsBean;
import com.jason.module_my.model.callback.OnRequestCallBack;
import com.jason.tools.http.BaseObserver;
import com.jason.tools.http.RetrofitManager;
import com.jason.tools.http.RxSchedulers;
import com.jason.tools.http.bean.JuheBaseResult;
import java.util.List;

/**
 * Created by jason on 2018/12/20.
 */

public class DetailsRepository {

    public void requestDetails(String id, final OnRequestCallBack<List<DetailsBean>> callBack){
        RetrofitManager.getInstance()
                .getRetrofit()
                .create(ApiService.class)
                .getDetails(Config.HISTORY_KEY,id)
                .compose(RxSchedulers.<JuheBaseResult<List<DetailsBean>>>compose())
                .subscribe(new BaseObserver<List<DetailsBean>>() {
                    @Override
                    public void onSuccess(List<DetailsBean> data) {
                        if (callBack!=null) callBack.getDataSuccess(data);
                    }

                    @Override
                    public void onFailed(String msg) {
                        if (callBack!=null) callBack.getDataFailed(msg);
                    }
                });
    }
}

File: module_my/src/main/java/com/jason/module_my/model/repository/DetailsRepository.kt
package com.jason.module_my.model.repository

import com.jason.module_my.http.ApiService
import com.jason.module_my.model.bean.DetailsResult
import com.jason.module_my.model.callback.OnRequestCallBack
import com.jason.tools.http.BaseObserver
import com.jason.tools.http.RetrofitManager
import com.jason.tools.http.RxSchedulers
import com.jason.tools.http.bean.JuheBaseResult

class DetailsRepository {

    fun requestDetails(id: String, callBack: OnRequestCallBack<List<DetailsResult>>?) {
        RetrofitManager.getInstance()
                .retrofit
                .create(ApiService::class.java)
                .getDetails(ApiService.HISTORY_KEY, id)
                .compose(RxSchedulers.compose<JuheBaseResult<List<DetailsResult>>>())
                .subscribe(object : BaseObserver<List<DetailsResult>>() {
                    override fun onSuccess(data: List<DetailsResult>) {
                        callBack?.getDataSuccess(data)
                    }

                    override fun onFailed(msg: String) {
                        callBack?.getDataFailed(msg)
                    }
                })

    }
}